core-bundle/src/Routing/Page/PageRouteFactory.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing\Page;
+
+use Contao\PageModel;
+use Symfony\Component\Routing\Exception\RouteNotFoundException;
+use Symfony\Component\Routing\Route;
+
+class PageRouteFactory
+{
+    /**
+     * @var array
+     */
+    private $routeConfigs = [];
+
+    /**
+     * @var array<PageRouteProviderInterface>
+     */
+    private $routeProviders = [];
+
+    public function createRoute(PageModel $pageModel, $content = null): Route
+    {
+        if (null !== ($route = $this->getRouteFromProvider($pageModel, $content))) {
+            return $route;
+        }
+
+        $route = new PageRoute($pageModel, $this->routeConfigs[$pageModel->type] ?? []);
+        $route->setContent($content);
+
+        return $route;
+    }
+
+    public function createRouteWithParameters(PageModel $pageModel, string $parameters = '', $content = null): Route
+    {
+        if (null !== ($route = $this->getRouteFromProvider($pageModel, $content))) {
+            return $route;
+        }
+
+        $route = $this->getRouteFromConfig($pageModel, $content);
+        $route->setPath(sprintf('/%s{parameters}', $pageModel->alias ?: $pageModel->id));
+        $route->setDefault('parameters', $parameters);
+        $route->setRequirement('parameters', $pageModel->requireItem ? '/.+' : '(/.+)?');
+
+        return $route;
+    }
+
+    public function add(string $type, RouteConfig $config, PageRouteProviderInterface $routeProvider = null): self
```

Comment:
This seems a bit strange to me. Do we ever need to have that changed at runtime? `add` sounds like I can add a `PageRoute` because this is a  `PageRouteFactory`. But I'm actually adding a new config which I can then use to `create...` from. So here we have a naming issue.

Reply:
I have added a `PageRegistry` (similar to the `FragmentRegistry`) that handles this instead in edd253ef7a20de9fe1351499bdf1a48ad5f9346a